### PR TITLE
Add Support for LPAD

### PR DESCRIPTION
**Pull request by Laneware**
## Description

Added support for LPAD

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lpad/config.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 Laneware Peripherals
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 3
+#define MATRIX_COLS 3
```
Comment (waffle87) [inline keyboards/lpad/config.h:10]:
```suggestion
```

File: keyboards/lpad/config.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+/*ENCODER*/
+#define ENCODERS_PAD_A { F0 }
+#define ENCODERS_PAD_B { F1 }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (waffle87) [inline keyboards/lpad/config.h:25]:
```suggestion
```

File: keyboards/lpad/info.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "maintainer": "qmk", 
+    "usb": {
+        "vid": "0x4C50",
+        "pid": "0x2222",
+        "device_version": "0.0.1"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
+    "layouts":{
+      "LAYOUT_default":{
```
Comment (waffle87) [inline keyboards/lpad/info.json:14]:
```suggestion
      "LAYOUT": {
```

File: keyboards/lpad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2023 Laneware Peripherals
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_default(
```
Comment (waffle87) [inline keyboards/lpad/keymaps/default/keymap.c:8]:
```suggestion
	[0] = LAYOUT (
```

File: keyboards/lpad/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+	[2] = LAYOUT_default(
+		KC_TRNS, KC_TRNS, 
+		KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[3] = LAYOUT_default(
+		KC_TRNS, KC_TRNS, 
+		KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS),};
```
Comment (waffle87) [inline keyboards/lpad/keymaps/via/keymap.c:26]:
```suggestion
	[0] = LAYOUT (
		KC_MPLY, KC_MUTE,  
		KC_MNXT,  
		KC_MPRV,  KC_LEFT,    KC_RIGHT),    

	[1] = LAYOUT (
		KC_TRNS, KC_TRNS, 
		KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS),

	[2] = LAYOUT (
		KC_TRNS, KC_TRNS, 
		KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS),

	[3] = LAYOUT (
		KC_TRNS, KC_TRNS, 
		KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS),
};
```

File: keyboards/lpad/lpad.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+/* ┌───┐
+ * │000│ ┌───┐
+ * ├───┤ │001│
+ * │100│ └───┘
+ * ├───┼───┬───┐
+ * │200│201│202│
+ * └───┴───┴───┘
+ */
+
+#define LAYOUT_default( \
```
Comment (waffle87) [inline keyboards/lpad/lpad.h:17]:
```suggestion
#define LAYOUT ( \
```

File: keyboards/lpad/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# LPAD
+
+![LPAD](https://i.imgur.com/bQgqZC6.jpg)
```
Comment (waffle87) [inline keyboards/lpad/readme.md:3]:
```suggestion
![LPAD](https://i.imgur.com/bQgqZC6h.jpg)
```

File: keyboards/lpad/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+A Macro Keyboard with a RHS rotary encoder, made and sold by Laneware Peripherals.
+
+* Keyboard Maintainer: [Laneware Peripherals](https://github.com/laneware)
+* Hardware Supported: LPAD
+* Hardware Availability: [Laneware Peripherals](https://lanewareperipherals.com/), 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lpad:default
```
Comment (waffle87) [inline keyboards/lpad/readme.md:13]:
```suggestion
    make lpad:default

Flashing example for this keyboard:

    make lpad:default:flash

```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/lpad/config.h [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+#define MATRIX_ROWS 3
+#define MATRIX_COLS 3
+
+/*ENCODER*/
+#define ENCODERS_PAD_A { F0 }
+#define ENCODERS_PAD_B { F1 }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
```
Comment (waffle87) [inline keyboards/lpad/config.h:22]:
```suggestion
/* key matrix pins */
#define MATRIX_ROW_PINS { E6, B7, D0}
#define MATRIX_COL_PINS { D3, D4, D6 }

/*ENCODER*/
#define ENCODERS_PAD_A { F0 }
#define ENCODERS_PAD_B { F1 }
```

File: keyboards/lpad/info.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    "manufacturer": "Laneware Peripherals",
+    "url": "", 
+    "maintainer": "qmk", 
+    "usb": {
+        "vid": "0x4C50",
+        "pid": "0x2222",
+        "device_version": "0.0.1"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/lpad/info.json:12]:
```suggestion
    "bootloader": "atmel-dfu",
    "diode_direction": "COL2ROW",
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by Laneware** [comment]
@waffle87 can I request some help on this one please? Checks were passing before requested updates, should I change them back?

File: keyboards/lpad/info.json [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    "diode_direction": "COL2ROW",
```
Comment (drashna) [inline keyboards/lpad/info.json:13]:
```suggestion
    "matrix_pins": {
        "cols": ["F5", "E6", "D0"],
        "rows": ["D3", "D4", "D6"]
    },
    "diode_direction": "COL2ROW",
    "encoder": {
        "rotary": [
            {
                "pin_a": "F0",
                "pin_b": "F1"
            }
        ]
    },
```

File: keyboards/lpad/config.h [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+#pragma once
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, B7, D0}
+#define MATRIX_COL_PINS { D3, D4, D6 }
+
+/*ENCODER*/
+#define ENCODERS_PAD_A { F0 }
+#define ENCODERS_PAD_B { F1 }
```
Comment (drashna) [inline keyboards/lpad/config.h:12]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lpad/config.h
Comment (drashna) [inline keyboards/lpad/config.h:1]: This file could probably be removed completely, as locking switch support is uncommon, and that's the only setting present here.

**Review by drashna — APPROVED** [review]

File: keyboards/lpad/info.json [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        "rotary": [
+            {
+                "pin_a": "F0",
+                "pin_b": "F1"
+            }
+        ]
+    },
+    "layouts":{
+      "LAYOUT": {
+        "layout": [{"label":"Play", "x":0, "y":0}, {"label":"Mute", "x":1.5, "y":0.5}, {"label":"Next", "x":0, "y":1}, {"label":"Prev", "x":0, "y":2}, {"label":"Left", "x":1, "y":2}, {"label":"Right", "x":2, "y":2}]
```
Comment (waffle87) [inline keyboards/lpad/info.json:28]:
```suggestion
                { "label": "Play", "matrix": [0, 0], "w": 1, "x": 0, "y": 0 },
                { "label": "Mute", "matrix": [0, 1], "w": 1, "x": 1.5, "y": 0.5 },
                { "label": "Next", "matrix": [1, 0], "w": 1, "x": 0, "y": 1 },
                { "label": "Prev", "matrix": [2, 0], "w": 1, "x": 0, "y": 2 },
                { "label": "Left", "matrix": [2, 1], "w": 1, "x": 1, "y": 2 },
                { "label": "Rght", "matrix": [2, 2], "w": 1, "x": 2, "y": 2 }
```

File: keyboards/lpad/lpad.c [hunk 14]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2023 Laneware Peripherals
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "lpad.h"
```
Comment (waffle87) [inline keyboards/lpad/lpad.c:4]:
```suggestion
#include "quantum.h"
```

File: keyboards/lpad/lpad.h
Comment (waffle87) [inline keyboards/lpad/lpad.h:1]: This file can be removed with above suggestions.

**Review by waffle87 — CHANGES_REQUESTED** [review]
Perplexed why this doesn't build with the current layout macro in `lpad.h`, but it belongs in `info.json` anyway so

**Review by waffle87 — APPROVED** [review]

**Pull request merged**